OpenDreamRuntime/Procs/DMOpcodeHandlers.cs

```diff
@@ -1860,6 +1860,117 @@ private static void HandleSuffixPronoun(ref StringBuilder formattedString, ReadO
+                            color2 = arg.Value;
+                            space = ColorHelpers.ColorSpace.RGB;
+                        } else if (name.StartsWith("b", StringComparison.InvariantCultureIgnoreCase) && color3 == default) {
+                            color3 = arg.Value;
+                            space = ColorHelpers.ColorSpace.RGB;
+                        } else if (name.StartsWith("h", StringComparison.InvariantCultureIgnoreCase) && color1 == default) {
+                            color1 = arg.Value;
+                            space = ColorHelpers.ColorSpace.HSV;
+                        } else if (name.StartsWith("s", StringComparison.InvariantCultureIgnoreCase) && color2 == default) {
+                            color2 = arg.Value;
+                            space = ColorHelpers.ColorSpace.HSV;
+                        } else if (name.StartsWith("v", StringComparison.InvariantCultureIgnoreCase) && color3 == default) {
+                            color3 = arg.Value;
+                            space = ColorHelpers.ColorSpace.HSV;
+                        } else if (name.StartsWith("a", StringComparison.InvariantCultureIgnoreCase) && a == default)
+                            a = arg.Value;
+                        else if (name == "space" && space == default)
+                            space = (ColorHelpers.ColorSpace)(int)arg.Value.UnsafeGetValueAsFloat();
+                        else
+                            throw new Exception($"Invalid or double arg \"{name}\"");
+                    }
+                }
+
+                if (color1 == default)
+                    throw new Exception("Missing first component");
+                if (color2 == default)
+                    throw new Exception("Missing second color component");
+                if (color3 == default)
+                    throw new Exception("Missing third color component");
+            } else {
+                state.Push(DreamValue.Null);
+                return ProcStatus.Continue;
+            }
+
+            int color1Value = (int)color1.UnsafeGetValueAsFloat();
+            int color2Value = (int)color2.UnsafeGetValueAsFloat();
+            int color3Value = (int)color3.UnsafeGetValueAsFloat();
+            byte aValue = (byte)Math.Clamp((int)a.UnsafeGetValueAsFloat(), 0, 255);
+            Color color;
+
+            switch (space) {
+                case ColorHelpers.ColorSpace.RGB: {
+                    byte r = (byte)Math.Clamp(color1Value, 0, 255);
+                    byte g = (byte)Math.Clamp(color2Value, 0, 255);
+                    byte b = (byte)Math.Clamp(color3Value, 0, 255);
+
+                    color = new Color(r, g, b, aValue);
+                    break;
+                }
+                case ColorHelpers.ColorSpace.HSV: {
+                    // TODO: Going beyond the max defined in the docs returns a different value. Don't know why.
+                    float h = Math.Clamp(color1Value, 0, 360) / 360f;
```

Comment:
is it maybe just wrapping? HSV is basically polar co-ordinates. Try mod?

Reply:
That's not it. s=101 and s=1 give two different results:
```dm
world.log << rgb(h=180, s=101, v=50)
world.log << rgb(h=180, s=1, v=50)
```
```
#008080
#7e8080
```